packages/grafana-runtime/src/services/dashboard-srv/types.ts

```diff
@@ -0,0 +1,123 @@
+import { PanelModel as BaseCorePanelModel } from '@grafana/data';
+import { GridPos } from '@grafana/schema';
+
+// Middle types to glue Public Plugin APIs types with internal Grafana types
+export interface CorePanelModel extends BaseCorePanelModel {
+  gridPos: GridPos;
+  refresh: () => void;
+}
+
+export interface CoreDashboardModel {
+  panels: CorePanelModel[];
+  uid: string;
+  title: string;
+  updatePanels: (panels: CorePanelModel[]) => unknown;
+}
+
+export interface CoreDashboardSrv {
+  dashboard?: CoreDashboardModel;
+}
+
+// Public types
+
+/**
+ * Used to interact with the current Grafana dashboard.
+ *
+ * @public
+ */
+export interface PluginsAPIDashboardSrv {
+  /**
+   * @deprecated use `getCurrentDashboard()` instead
+   */
+  dashboard?: PluginsAPIDashboardModel;
+  /**
+   * Represents the currently loaded dashboard. Returns 'undefined' if no dashboard is loaded.
+   */
+  getCurrentDashboard(): PluginsAPIDashboardModel | undefined;
+}
+
+/**
+ * Used to interact with the current Grafana dashboard.
+ * @public
+ */
+export interface PluginsAPIDashboardModel {
+  /**
+   * The uid of the dashboard
+   * @remarks
+   * Note: The old `id` attribute is deprecated and therefore not included in the public API.
+   * @readonly
+   */
+  uid: string;
+  /**
+   * The title of the dashboard
+   * @readonly
+   */
+  title: string;
+  /**
+   * @readonly
+   * @deprecated use `getPanels()`
+   */
+  panels: PluginsAPIPanelModel[];
+
+  /**
+   * Returns the panels within the current dashboard.
+   *
+   * Note: Modifications to the panels are not immediately visible on the dashboard and are not automatically saved.
+   *
+   * Users must manually save the dashboard to persist any changes.
+   */
+  getPanels(): PluginsAPIPanelModel[];
+
+  /**
+   * Replaces the existing panels on the current dashboard with the provided ones.
+   *
+   * To retrieve current panels, refer to `getPanels()`.
+   *
+   * Note: While panel changes are instantly reflected on the dashboard, they are not automatically saved.
+   *
+   * Users must manually save the dashboard to persist any modifications.
+   *
+   * @public
+   */
+  updatePanels(panels: PluginsAPIPanelModel[]): void;
```

Comment:
Could this function be exposed directly from the dashboardSrv instead of nesting them in a model?

Reply:
Similar to my previous answers. We avoid bloating dashboardSrv. 

Having the apis in hierarchical object doesn't make them harder to use and in a object oriented programming style makes more sense.